node/src/main/scala/coop/rchain/node/runtime/Setup.scala

```diff
@@ -106,6 +106,19 @@ object Setup {
 
       // Block storage
       blockStore <- KeyValueBlockStore(rnodeStoreManager)
+      blockKeyValueStore <- rnodeStoreManager
+                             .store("blocks")
+                             .map(
+                               _.toTypedStore[BlockHash, BlockMessage](
+                                 codecBlockHash,
+                                 codecBlockMessage
+                               )
+                             )
+      blockMetadataStore <- rnodeStoreManager.database[BlockHash, BlockMetadata](
+                             "block-metadata",
+                             codecBlockHash,
+                             codecBlockMetadata
+                           )
```

Comment:
Instead of directly working with underlying store used in BlockMetadataStore define function(s) which can access data you need from BlockMetadataStore class and supply these function to BlockReceiver. in that way only BlockReceiver should now know internal implementation for BlockMetadataStore.

Reply:
I [used](https://github.com/rchain/rchain/pull/3650/commits/7981e75498697da43ad7426abd00fde5b075baaf) little bit different approach. I just pass unvalidated and validated set of hashes.